packages/types/src/types/database-interface.ts

```diff
@@ -70,7 +70,7 @@ export interface DatabaseInterfaceSessions {
     token: string,
     connection: ConnectionInformations,
     extraData?: object
-  ): Promise<string>;
+  ): Promise<string | null>;
```

Comment:
@birkir why did you add null there?

Reply:
The Session can be null from the database getter methods. I tried to make it as flexible as possible but I really wanted to keep the constrain on User<>Session. So when TypeORM is used for the e2e test, we cannot guarantee that the result from `findSessionById` and `findSessionByToken` returns the ID.

User delete cascades related sessions.

My first design actually removed sessions as they were invalidated, (instead of soft delete), because I didn't really understand the need. I ended up reverting that for compatibility with redis and mongo.

Reply:
I see, removing session when invalidating them is a good idea, would be great to create an option for that :)

But this is for a create function, is it really possible that a create function return `null`?